File: sdk/phoenix-client-java/src/main/java/io/github/vishva_kalhara/phoenix/util/PhoenixException.java
package io.github.vishva_kalhara.phoenix.util;

/**
 * Custom exception class for handling errors within the Phoenix client.
 * <p>
 * This exception is used to encapsulate various errors that may occur
 * during API requests, network failures, or application logic errors.
 * </p>
 */
public class PhoenixException extends Exception {

    /**
     * Constructs a new {@code PhoenixException} with the specified error message.
     *
     * @param message The error message describing the cause of the exception.
     */
    public PhoenixException(String message) {
        super(message);
    }

    /**
     * Constructs a new {@code PhoenixException} with the specified cause.
     *
     * @param cause The underlying throwable that caused this exception.
     */
    public PhoenixException(Throwable cause) {

        super(cause);
    }
}

File: packages/phoenix-client-cs/Phoenix/Util/PhoenixException.cs
using System;

namespace Phoenix.Util
{
    public class PhoenixException : Exception
    {
        public PhoenixException(string message) : base(message)
        {
        }

        public PhoenixException(string message, Exception innerException) : base(message, innerException)
        {
        }
    }
}